### PR TITLE
Test  allocations

### PR DESCRIPTION
**Pull request by jaredpar**
Continuing to work on unnecessary allocations in our unit tests

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs [hunk 1]
```diff
@@ -12384,7 +12384,7 @@ void binaryOperator(string op, string leftType, string rightType, string expecte
 
                 if (expectedDiagnostics.Length == 0)
                 {
-                    CompileAndVerify(comp);
+                    CompileAndVerify(comp, emitOptions: EmitOptions.Default);
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs:12387]: By default `CompileAndVerify` embeds portable PDBs and the sheer number of tests here was causing that allocate 100MB of unnecessary `byte[]`.

File: src/Compilers/Test/Core/Platform/Desktop/Extensions.cs [hunk 2]
```diff
@@ -13,6 +13,7 @@
 using System.IO;
 using System.Linq;
 using System.Reflection;
+using System.Runtime.InteropServices;
```
Comment (jaredpar) [inline src/Compilers/Test/Core/Platform/Desktop/Extensions.cs:16]: This is a significant source of redundant `byte[]` allocations.

File: src/Compilers/Test/Core/TargetFrameworkUtil.cs [hunk 3]
```diff
@@ -113,13 +113,26 @@ public static class NetCoreApp
...
+        private static ImmutableArray<MetadataReference> s_references;
+
         /// <summary>
         /// This is the full set of references provided by default on the .NET Framework TFM
         /// </summary>
         /// <remarks>
         /// Need to special case tuples until we move to net472
         /// </remarks>
-        public static ImmutableArray<MetadataReference> References { get; } = [.. Net461.References.All, Net461.ExtraReferences.SystemValueTuple];
+        public static ImmutableArray<MetadataReference> References
```
Comment (jaredpar) [inline src/Compilers/Test/Core/TargetFrameworkUtil.cs:124]: There are a number of tests that call say `NetFramework.mscorlib`. When this was a static initializer that realized all these values when they didn't really need to be. This is only a savings in certain silces of our tests but it's a significant one in them.

File: src/Compilers/Test/Core/Platform/Desktop/Extensions.cs [hunk 4]
```diff
@@ -30,27 +31,27 @@ public static void AddArray<T>(this SerializationInfo info, string name, Immutab
             // we will copy the content into an array and serialize the copy
             // we could serialize element-wise, but that would require serializing
             // name and type for every serialized element which seems worse than creating a copy.
-            info.AddValue(name, value.IsDefault ? null : value.ToArray(), typeof(T[]));
+            info.AddValue(name, value.IsDefault ? null : ImmutableCollectionsMarshal.AsArray(value), typeof(T[]));
```
Comment (AlekseyTs) [inline src/Compilers/Test/Core/Platform/Desktop/Extensions.cs:34]: Is the comment above still accurate?

File: src/Compilers/Test/Core/Platform/Desktop/Extensions.cs [hunk 5]
```diff
@@ -30,27 +31,27 @@ public static void AddArray<T>(this SerializationInfo info, string name, Immutab
...
+            return ImmutableCollectionsMarshal.AsImmutableArray(array);
         }
 
         public static void AddByteArray(this SerializationInfo info, string name, ImmutableArray<byte> value)
         {
             // we will copy the content into an array and serialize the copy
             // we could serialize element-wise, but that would require serializing
             // name and type for every serialized element which seems worse than creating a copy.
-            info.AddValue(name, value.IsDefault ? null : value.ToArray(), typeof(byte[]));
+            info.AddValue(name, value.IsDefault ? null : ImmutableCollectionsMarshal.AsArray(value), typeof(byte[]));
```
Comment (AlekseyTs) [inline src/Compilers/Test/Core/Platform/Desktop/Extensions.cs:48]: Here as well

Comment (jaredpar) [reply src/Compilers/Test/Core/Platform/Desktop/Extensions.cs:48]: Think those comments are somewhat wrong even in the before case. Let me clean that up.

**Review by 333fred — APPROVED** [review]

**Review by AlekseyTs — COMMENTED** [review]
LGTM (commit 2)

**Pull request merged**